content/terraform/v1.12.x/docs/language/manage-sensitive-data.mdx

```diff
@@ -0,0 +1,287 @@
+  sensitive   = true
+}
+```
+
+</CodeBlockConfig>
+
+Terraform then redacts sensitive variables and outputs in the operation log of the Terraform CLI. For example, the `connection_string` output is redacted in the following apply summary:
+
+```shell-session
+Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+connection_string = (sensitive value)
+```
+
+Terraform also automatically treats any expression that references a sensitive variable or output as sensitive. In the following example, the `password` argument references the sensitive `var.database_password` in another resource:
+
+<CodeBlockConfig highlight="5">
+
+```hcl
+resource "aws_db_instance" "main" {
+  identifier = "my-database"
+  engine     = "postgres"
+  username   = var.db_username
+  password   = var.database_password
+}
+```
+
+</CodeBlockConfig>
+
+Terraform redacts the `password` argument from the `main` resource in plan and apply operations:
+
+<CodeBlockConfig highlight="3">
+
+```shell-session
+# aws_db_instance.main will be created
++ resource "aws_db_instance" "main" {
+    + password = (sensitive value)
+    + username = "admin"
+  }
+```
+
+</CodeBlockConfig>
+
+Terraform stores values with the `sensitive` argument in both state and plan files, and anyone who can access those files can access your sensitive values. Additionally, if you use the [`terraform output` CLI command](/terraform/cli/commands/output) with the `-json` or `-raw` flags, Terraform displays sensitive variables and outputs in plain text.
+
+If you do not want to store the values of your sensitive variables and outputs, refer to [Omit values from state and plan files](#omit-values-from-state-and-plan-files). If you have sensitive values in your state and plan files, we recommend [securing your state](#state-security-best-practices).
+
+## Hide values without storing them
+
+You can add the `sensitive` argument to variables and child module outputs that have the `ephemeral` argument to combine their benefits. The following `database_password` variable is omitted from state and plan files, and the value is redacted in the Terraform CLI and the HCP Terraform UI:
+
+<CodeBlockConfig highlight="4-5">
+
+```hcl
+variable "database_password" {
+  description = "Password for the database instance"
+  type        = string
+  sensitive   = true
+  ephemeral   = true
+}
+```
+
+</CodeBlockConfig>
+
+To learn more about the `ephemeral` argument, refer to [Omit values from state and plan files](#omit-values-from-state-and-plan-files).
+
+## Omit values from state and plan files
+
+Ephemeral values are available at the run time of an operation, but Terraform omits them from state and plan files. Because Terraform does not store ephemeral values, you must capture any generated values you want to preserve in another resource or output in your configuration.
+
+Terraform provides four ways to define ephemeral values in your configuration:
```

Comment:
```suggestion
You can use the following methods ways to define ephemeral values in your configuration:
```
I count three ways, btw

Reply:
Lol thank you, this is what I get for naming a number and changing my mind on formatting 😂